acceptance/builder_test.go

```diff
@@ -0,0 +1,418 @@
+					command := exec.Command(
+						"docker",
+						"run",
+						"--rm",
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+						builderImage,
+					)
+					output, err := command.CombinedOutput()
+					h.AssertNotNil(t, err)
+					expected := "failed to parse detect plan"
+					h.AssertStringContains(t, string(output), expected)
+				})
+			})
+
+			//TODO: check some output file for this case not except any error message
+			when("empty", func() {
+				it("success", func() {
+					h.DockerRunAndCopy(t,
+						containerName,
+						copyDir,
+						ctrPath("/layers"),
+						builderImage,
+						h.WithFlags(
+							"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+							"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/empty_plan.toml",
+							"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+						),
+					)
+					// check builder metadata.toml for success test
+					assertBuilderMetadata(t, filepath.Join(copyDir, "layers/config/metadata.toml"))
+				})
+			})
+
+			when("invalid", func() {
+				it("errors", func() {
+					command := exec.Command(
+						"docker",
+						"run",
+						"--rm",
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/wrong_plan.toml",
+						"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+						builderImage,
+					)
+					output, err := command.CombinedOutput()
+					h.AssertNotNil(t, err)
+					expected := "failed to parse detect plan: Near line"
+					h.AssertStringContains(t, string(output), expected)
+				})
+			})
+		})
+	})
+
+	when("determining the location of input files", func() {
+		// .../cmd/lifecycle/builder.go#Args
+		when("group.toml path is not specified", func() {
+			it("will look for group.toml in the provided layers directory", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_LAYERS_DIR=/layers/different_layer_dir_from_env",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+					builderImage,
+				)
+				_, err := command.CombinedOutput()
+				h.AssertNil(t, err)
+				//expected := "failed to read buildpack group: open /layers/group.toml"
+				//h.AssertStringContains(t, string(output), expected)
+			})
+		})
+
+		// .../cmd/lifecycle/builder.go#Args
+		when("plan.toml path is not specified", func() {
+			it("will look for plan.toml in the provided layers directory", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_LAYERS_DIR=/layers/different_layer_dir_from_env",
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					builderImage,
+				)
+				_, err := command.CombinedOutput()
+				h.AssertNil(t, err)
+				//expected := "failed to parse detect plan: open /layers/plan.toml"
```

Comment:
Same comment as https://github.com/buildpacks/lifecycle/pull/648/files#r672620621

Reply:
different buildpack is added. then metadata file is checked.